import sys
from datetime import datetime
import mysql.connector as mariadb

# user created library
import misc_util

class DataDB:
   def __init__(self, logspk ):
       self.conn   = ''
       self.cursor = ''
       self.logspk = logspk

       self.confquery = "INSERT INTO conference_data (CoSpace_id, Name, Autogenerated," \
                         " Uri, CallId, SecondaryUri, date_received, Ip, local_remote," \
                         "contract, month ) " \
                         "VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)"
       self.devquery  = "INSERT INTO device_data (TotalDevices, TotalUsers, EnhancedPlus," \
                         " Enhanced, TelePresence,CuwlStandard, Basic, Essential, Timestamp," \
                         " Elm, ElmLastContact, date_received, Ip, local_remote, contract, month) " \
                         "VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)"

       try:
          self.conn = mariadb.connect(database='Billing')
          self.cursor = self.conn.cursor()
       except mariadb.Error as db_error:
         self.logspk.error("CUCM/CMS billing database connection failure: %s " % db_error) 


   def load_data_db(self, xml_type, tmplis, serv_info):
        ''' Inserts the device/usage and conference room data into database '''

        if self.conn:
           if xml_type == 'conf_spaces':
              for tmpdic in tmplis[1:]:
                 try:
                     insert_data = (tmpdic['coSpace id'], tmpdic['name'], tmpdic['autogenerated'],
                                    tmpdic['uri'], tmpdic['callId'], tmpdic.get('secondaryUri',
                                    'None'), serv_info['currdate'], serv_info['ip_address'], 
                                    serv_info['local_or_remote'], serv_info['contract'], serv_info['month'])

                     self.cursor.execute(self.confquery, insert_data)
                     self.conn.commit()
                 except mariadb.Error as db_error:
                     misc_util.log_message(self.logspk, serv_info, 'Database inserting failure for conference data',
                                           'sys_error', db_error)

           elif xml_type == 'devices':
              tmpdic = tmplis[0]
              try:
                 # data to be inserted into database
                 insert_data = (tmpdic['TotalDevices'], tmpdic['TotalUsers'], tmpdic['EnhancedPlus'],
                                tmpdic['Enhanced'], tmpdic['TelePresence Room'], tmpdic['CUWL Standard'],
                                tmpdic['Basic'], tmpdic['Essential'], tmpdic['Timestamp'],
                                tmpdic['Elm'], tmpdic['ElmLastContact'], serv_info['currdate'], serv_info['ip_address'], 
                                serv_info['local_or_remote'], serv_info['contract'], serv_info['month'] )
                 self.cursor.execute(self.devquery, insert_data)
                 self.conn.commit()
              except mariadb.Error as db_error:
                 misc_util.log_message(self.logspk, serv_info, 'Database inserting failure for device/usage data',
                                           'sys_error', db_error)
              

   def close_db_access(self):
       if self.conn:
          self.conn.close()
